packages/medusa/src/api-v2/store/payment-collections/[id]/payment-sessions/route.ts

```diff
@@ -0,0 +1,64 @@
+import { createPaymentSessionsWorkflow } from "@medusajs/core-flows"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPaymentModuleService } from "@medusajs/types"
+import { remoteQueryObjectFromString } from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../types/routing"
+import { defaultStorePaymentCollectionFields } from "./query-config"
+import { StorePostPaymentCollectionsPaymentSessionReq } from "./validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<StorePostPaymentCollectionsPaymentSessionReq>,
+  res: MedusaResponse
+) => {
+  const { id } = req.params
+  const { context, provider_id, data } = req.body
+
+  // If the customer is logged in, we auto-assign them to the payment collection
+  if (req.auth?.actor_id) {
```

Comment:
Should this happen in some middleware instead?

Reply:
I could introduce a generic middleware `maybeAttachCustomerIdFromAction` for reusability purposes. Wdyt?

Reply:
@olivermrbl yeah I think that would make a lot more sense, eg. `attachActorToCustomer`. I am a bit uncertain about the exact logic we want here, but would it even make sense if it's part of the auth middleware?

Reply:
Went with a slightly simpler approach in f0c9aa0f100cbc3153de14e2c. I checked other routes that use the actor ID and there are no commonalities yet, so a middleware seems overkill. Some use `customer.id`, others `created_by`, and here `customer_id`.

Reply:
Went with a slightly simpler approach in f0c9aa0f100cbc3153de14e2c for cart creation and kept this as is. I checked other routes that use the actor ID and there are no commonalities yet, so a middleware seems overkill. Some use `customer.id`, others `created_by`, and here `customer_id`.